streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -1320,6 +1320,19 @@ public void shouldUseDefaultStateStoreCacheMaxBytesConfigWhenNoConfigIsSet() {
         assertEquals(getTotalCacheSize(config), 10 * 1024 * 1024);
     }
 
+    @Test
+    public void testStateStoreMaxUncommittedBytesShouldAllowUnbounded() {
+        props.put(StreamsConfig.STATESTORE_UNCOMMITTED_MAX_BYTES_CONFIG, -1);
+        final StreamsConfig config = new StreamsConfig(props);
+        assertEquals(Long.valueOf(-1), config.getLong(StreamsConfig.STATESTORE_UNCOMMITTED_MAX_BYTES_CONFIG));
+    }
+
+    @Test
+    public void shouldUseDefaultStateStoreMaxUncommittedBytesConfigWhenNoConfigIsSet() {
+        final StreamsConfig config = new StreamsConfig(props);
+        assertEquals(Long.valueOf(64 * 1024 * 1024), config.getLong(StreamsConfig.STATESTORE_UNCOMMITTED_MAX_BYTES_CONFIG));
+    }
+
```

Comment:
With a value of `0`, it would trigger an "early" commit whenever transaction buffers contain any data. If we decide to disallow this, then we need to determine where we draw the line, because a value of `1` or `2`, for example, would likely yield the same result.